src/OrleansRuntime/Scheduler/WorkerPoolThread.cs

```diff
@@ -328,7 +330,7 @@ internal void CheckForLongTurns()
             // Note that we only do this if the current load is reasonably low and the current thread
             // count is reasonably small.
             if (!pool.ShouldInjectWorkerThread ||
-                (Silo.CurrentSilo == null || !(Silo.CurrentSilo.Metrics.CpuUsage < MAX_CPU_USAGE_TO_REPLACE))) return;
+                !(this.performanceMetrics.CpuUsage < MAX_CPU_USAGE_TO_REPLACE)) return;
```

Comment:
should this take into account that performanceMetrics can be null? (since I saw some path that you explicitly new up with null, although not sure whether it will hit this line in that path).